packages/recommend-core/src/personalization/getPersonalizationFilters.ts

```diff
@@ -0,0 +1,54 @@
+import { getAffinities } from './getAffinities';
+import { getStrategy } from './getStrategy';
+
+type GetPersonalizationFilters = {
+  userToken: string;
+  region: string;
+  apiKey: string;
+  appId: string;
+};
+
+export const getPersonalizationFilters = async ({
+  userToken,
+  region,
+  apiKey,
+  appId,
+}: GetPersonalizationFilters) => {
+  if (!region || !apiKey || !appId) {
+    throw new Error(
+      `[Algolia Recommend] parameters 'region', 'apiKey' and 'appId' are required to enable personalization.`
+    );
+  }
+  if (!userToken) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `[Algolia Recommend] parameter 'userToken' is required to enable personalization.`
+    );
+    return [];
+  }
+
+  try {
+    const [affinities, strategy] = await Promise.all([
+      getAffinities({ userToken, apiKey, appId, region }),
+      getStrategy({ apiKey, appId, region }),
+    ]);
+
+    const FALLBACK_WEIGHT = 100;
+    const facetsScoringMap = new Map(
+      strategy.facetsScoring.map((value) => [value.facetName, value.score])
+    );
+
+    const optionalFilters = Object.entries(affinities.scores).flatMap(
+      ([facet, values]) =>
+        Object.entries(values).map(([value, score]) => {
+          const weight = facetsScoringMap.get(facet) ?? FALLBACK_WEIGHT;
+          const weightedScore = Math.floor(score * (weight / 100));
+          return `${facet}:${value}<score=${weightedScore}>`;
+        })
+    );
```

Comment:
Is the engine able to receive `X` filters? Isn't there a limit where it would throw?

Reply:
If there is a reasonable limit, i haven't seen it yet.
I tired some queries with ~25 filters and the engine responded fine.

Before adding a limit, I guess we can look at strategies and see how many attributes are used in maximum cases.